assets/src/edit-story/app/font/actions/useLoadFontFiles.js

```diff
@@ -25,38 +25,86 @@ import { useCallback } from 'react';
 import cleanForSlug from '../../../utils/cleanForSlug';
 import getGoogleFontURL from '../../../utils/getGoogleFontURL';
 
+/**
+ * This is a utility ensure that Promise.all return ONLY when all promises are processed.
+ *
+ * @param {Promise} promise Promise to be processed
+ * @return {Promise} Return a rejected or fulfilled Promise
+ */
+const reflect = (promise) => {
+  return promise.then(
+    (v) => ({ v, status: 'fulfilled' }),
+    (e) => ({ e, status: 'rejected' })
+  );
+};
+
 function useLoadFontFiles() {
   /**
    * Adds a <link> element to the <head> for a given font in case there is none yet.
    *
    * Allows dynamically enqueuing font styles when needed.
    *
-   * @param {string} name Font name.
+   * @param {Array} props An array of fonts properties to create a valid FontFaceSet to inject and preload a font-face
+   * @return {Promise} Returns fonts loaded promise
    */
-  const maybeEnqueueFontStyle = useCallback(({ family, service, variants }) => {
-    if (!family || service !== 'fonts.google.com') {
-      return;
-    }
+  const maybeEnqueueFontStyle = useCallback((fonts) => {
+    return Promise.all(
+      fonts
+        .map(
+          async ({
+            font: { family, service, variants },
+            fontWeight,
+            fontStyle,
+          }) => {
+            if (!family || service !== 'fonts.google.com') {
+              return null;
+            }
+
+            const handle = cleanForSlug(family);
+            const elementId = `${handle}-css`;
+            const fontFaceSet = `
+              ${fontStyle || ''} ${fontWeight || ''} 0 '${family}'
+            `.trim();
 
-    const handle = cleanForSlug(family);
-    const id = `${handle}-css`;
-    const element = document.getElementById(id);
+            const hasFontLink = () => {
+              return document.getElementById(elementId);
+            };
 
-    if (element) {
-      return;
-    }
+            const appendFontLink = () => {
+              return new Promise((resolve, reject) => {
+                const src = getGoogleFontURL([{ family, variants }]);
```

Comment:
Did you confirm the `variants` format with @swissspidy? It's a bit cryptic in Google Fonts, so I always stumble over it.

Reply:
Nope. Actually, I only passed this variable down as a potential value used inside `getGoogleFontURL`.  Is there some special need to process the data inside `variants`?

Reply:
I think if you just pass them all - it's all good. If you pass only partial variants, the `getGoogleFontURL` will selectively only load few styles and you won't be able to use `fonts.load()` correctly at all. I think just passing them all is totally fine. But /cc @swissspidy just in case. I don't know if there's a way to just tell Google Fonts to load all variants no matter what. But I'm pretty happy if we just do it explicitly.